modules/gke/variables.tf

```diff
@@ -0,0 +1,31 @@
+variable "namespace" {
+  description = "Namespace for the GKE cluster"
+  type        = string
+}
+
+variable "service_account" {
+  description = "The service account associated with the GKE cluster instances to host Ctrlplane."
+  type        = object({ email = string })
+}
+
+variable "network" {
+  description = "Google Compute Engine network to which the cluster is connected."
+  type        = object({ self_link = string })
+}
+
+variable "subnetwork" {
+  description = "Google Compute Engine subnetwork in which the cluster's instances are launched."
+  type        = object({ self_link = string })
+}
+
+variable "machine_type" {
+  description = "The machine type for the cluster"
+  type        = string
+  default     = "n2-standard-4"
```

Comment:
looked at the comparison chart and n2 is listed as "general purpose", I don't think we need the super high performant ones